examples/bpi-monolith-test-suite/bpi/components/vsm/vsm.ts

```diff
@@ -0,0 +1,43 @@
+import { Agreement } from "../storage/agreement";
+import { Transaction } from "../transactions/transaction";
+import { IStorageComponent } from "../storage/storage.interface";
+import { IWorkgroupComponent } from "../workgroup/workgroup.interface";
+import { IVirtualStateMachineComponent } from "./vsm.interface";
+
+
+export class VirtualStateMachine implements IVirtualStateMachineComponent {
+    storageComponent: IStorageComponent;
+    workgroupComponent: IWorkgroupComponent;
+
+    constructor(storageComponent: IStorageComponent, workgroupComponent: IWorkgroupComponent) {
+        this.storageComponent = storageComponent;
+        this.workgroupComponent = workgroupComponent;
+    }
+
+    executeTransaction(transaction: Transaction): string {
+        // TODO: Poor man's deep copy - change
+        const latestAgrementState = this.storageComponent.getAgreementState();
+        const originalAgreementState = JSON.parse(JSON.stringify(latestAgrementState));
+
+        const workgroup = this.workgroupComponent.getWorkgroupById(transaction.workgroupId);
+        const workstep = workgroup.getWorkstepById(transaction.workstepId);
```

Comment:
ok. so the relationship is workgroup -> workflow -> workstep. a workstep is never directly attached to a workgroup unless workflow = workstep. @ognjenkurtic

Reply:
Yep, we just simplified for the first use-case. Will extend with upcoming work.